src/XMakeBuildEngine/Microsoft.Build.csproj

```diff
@@ -13,10 +13,10 @@
   </PropertyGroup>
   <!-- Default configurations to help VS understand the configurations -->
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
-    <DefineConstants>TRACE;DEBUG;STANDALONEBUILD;BUILD_ENGINE</DefineConstants>
+    <DefineConstants>TRACE;DEBUG;STANDALONEBUILD;BUILD_ENGINE;$(DefineConstantsMsbuildDebugger)</DefineConstants>
   </PropertyGroup>
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' ">
-    <DefineConstants>TRACE;STANDALONEBUILD;BUILD_ENGINE</DefineConstants>
+    <DefineConstants>TRACE;STANDALONEBUILD;BUILD_ENGINE;$(DefineConstantsMsbuildDebugger)</DefineConstants>
```

Comment:
BUILD_ENGINE is a confusing case, @AndyGerlicher.  It controls compilation for some of the shared source files that have things like

```C#
#if BUILD_ENGINE
namespace Microsoft.Build.BackEnd
#else
namespace Microsoft.Build.Utilities
#endif
{
    /// <summary>
    /// Helper logging class - contains all the logging methods used by tasks.
    /// A TaskLoggingHelper object is passed to every task by MSBuild. For tasks that derive
    /// from the Task class, it is provided in the Log property.
    /// This class is thread safe: tasks can log from any threads.
    /// </summary>
#if BUILD_ENGINE
    internal
#else
    public
#endif
 class TaskLoggingHelper : MarshalByRefObject
```

So we have to keep defining it only in Microsoft.Build.

@ValMenn is it possible to change this proj file so it just always appends TRACE;STANDALONE;BUILD_ENGINE to the list instead of adding it in this way?